### PR TITLE
#9 Feature:Create roles and permisions

### PR DESCRIPTION
**Pull request by Juleshb**
What does this PR do?

Create roles and set permissions

Description of Task to be completed?

-The Administrator should be able to create, update, and delete roles.
-The administrator should be able to set and update role's permissions.
-The Access control has been implemented.
-Appropriate error messages are displayed on unsuccessful attempts

How should this be manually tested?

-Clone the repo
-checkout to branch Fe-Create-roles-and-set-permisions
-Run npm install to install modules
-Run npm run dev:start to start the development server

Go to your postman
-Open postman and run POST http://127.0.0.1:3000/roles/api/v1/createRole
-Go to Body in your postman select JSON
Screen Shot 2022-07-13 at 14 40 12
Screen Shot 2022-07-13 at 14 47 40

### CODE REVIEW

**Review by songa1 — COMMENTED** [review]
We don't have migration for the Permissions and Roles. Please add Migrations.

**Review by songa1 — APPROVED** [review]

**Review by Rukundo725 — COMMENTED** [review]
There is no relationship between the user's table and one of the roles and permissions. Try to create their associations

**Review by Stevenkwizera06 — APPROVED** [review]

File: src/models/Permissions.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+module.exports = (sequelize, DataTypes) => {
+  // eslint-disable-next-line require-jsdoc
+  class Permissions extends Model {
+    /**
+     * Helper method for defining associations.
+     * This method is not a part of Sequelize lifecycle.
+     * The `models/index` file will call this method automatically.
+     */
+     static associate({Roles}) {
```
Comment (Rukundo725) [inline src/models/Permissions.js:11]: I think we have many_to_many relationships on creating roles and permissions, though you have used one_to_many. check well the Trello board here https://trello.com/c/zRlfxZP8/9-create-roles-and-set-permissions

File: src/models/roles.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+module.exports = (sequelize, DataTypes) => {
+  // eslint-disable-next-line require-jsdoc
+  class Roles extends Model {
+    /**
+     * Helper method for defining associations.
+     * This method is not a part of Sequelize lifecycle.
+     * The `models/index` file will call this method automatically.
+     */
+     static associate({Permissions}) {
+      this.hasMany(Permissions, { foreignKey: "roleId" });
```
Comment (Rukundo725) [inline src/models/roles.js]: I think we have many_to_many relationships on creating roles and permissions, though you have used one_to_many. check well the Trello board here https://trello.com/c/zRlfxZP8/9-create-roles-and-set-permissions

**Review by Rukundo725 — COMMENTED** [review]
I think we have many_to_many relationships on creating roles and permissions, though you have used one_to_many. check well the Trello board here https://trello.com/c/zRlfxZP8/9-create-roles-and-set-permissions

File: src/models/Permissions.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      this.belongsTo(Roles, { foreignKey: "roleId"});
+      // define association here
+    }
```
Comment (Rukundo725) [inline src/models/Permissions.js:16]: I think we have many_to_many relationships on creating roles and permissions, though you have used one_to_many. check well the Trello board here https://trello.com/c/zRlfxZP8/9-create-roles-and-set-permissions

**Review by Rukundo725 — APPROVED** [review]

**Pull request merged**